Makefile

```diff
@@ -1,22 +1,92 @@
+#! /usr/bin/make -f
+
+# Project variables.
+VERSION := development
+PROJECT_NAME := $(shell basename "$(PWD)")
 DATE := $(shell date '+%Y-%m-%dT%H:%M:%S')
-VERSION = development
 HEAD = $(shell git rev-parse HEAD)
 LD_FLAGS = -X github.com/tendermint/starport/starport/internal/version.Version='$(VERSION)' \
 	-X github.com/tendermint/starport/starport/internal/version.Head='$(HEAD)' \
 	-X github.com/tendermint/starport/starport/internal/version.Date='$(DATE)'
 BUILD_FLAGS = -mod=readonly -ldflags='$(LD_FLAGS)'
 
+# Go related variables.
+GOBASE := $(shell pwd)
+GOBIN := $(GOBASE)/dist
+GOCMD := $(GOBASE)/starport/cmd/starport
+GOCILINT := $(GOPATH)/bin/golangci-lint
+GORELEASER := $(GOPATH)/bin/goreleaser
+STARPORT := $(GOPATH)/bin/starport
+
+# For go 1.13-1.15 compatibility
+# https://maelvls.dev/go111module-everywhere/
+export GO111MODULE=on
+
+## install: Install de binary.
 install:
 	@echo Installing Starport...
 	@go install $(BUILD_FLAGS) ./...
 	@starport version
 
+## build-binary: Build the binary.
+build-binary: clean
+	@echo Building Starport...
+	@go build $(BUILD_FLAGS) -o $(GOBIN)/$(PROJECT_NAME) $(GOCMD)
+
+$(STARPORT): install
+
+$(GORELEASER):
+	@echo Installing goreleaser...
+	@go install github.com/goreleaser/goreleaser@latest
+
+## goreleaser: Build the binaries with Goreleaser.
+goreleaser: clean $(GORELEASER)
+	@echo Building with goreleaser...
+	@$(GORELEASER) --snapshot --skip-publish
+
+## clean: Clean build files. Runs `go clean` internally.
+clean:
```

Comment:
I believe this is a Go spesific thing, not directly related to Starport.

Reply:
it's for remove the files generated by the build, but we can remove it, and the build command

Reply:
I'm in favor of keeping the Makefile simple as much as possible. I almost never use the `go clean` command for Starport, but if you think it's needed I think we can keep it.